git-rebase-via-merge.sh

```diff
@@ -127,35 +129,47 @@ init(){
   done
 }
 
-get_any_changed_files(){
+get_any_changed_files() {
   git status --porcelain --ignore-submodules=dirty | cut -c4-
 }
 
-get_unstaged_files(){
+get_unstaged_files() {
   git status --porcelain --ignore-submodules=dirty | grep -v "^. " | cut -c4-
 }
 
-merge_conflicts_present(){
+get_files_with_conflict_markers() {
+  git diff --check | cat
+}
+
+merge_conflicts_present() {
   file_merge="$(git rev-parse --show-toplevel)/.git/MERGE_HEAD"
-  [ -e $file_merge ]
+  [ -e "$file_merge" ]
+}
+
+rebase_conflicts_present() {
+  [[ $(git diff --name-only --diff-filter=U --relative) ]]
 }
 
-get_hash(){
+get_hash() {
   git rev-parse --short "$1" || true
 }
 
-show_commit(){
+show_commit() {
   git log -n 1 --pretty=format:"%<(20)%an | %<(14)%ar | %s" "$1"
 }
 
-fix_merge_conflicts(){
-  while true
-  do
+fix_merge_conflicts() {
+  while true; do
     echo "Fix all conflicts in the following files, stage all the changes and type 'c':"
     get_unstaged_files
+    echo
+
+    echo "List of conflict markers:"
+    get_files_with_conflict_markers
+    echo
```

Comment:
Sometimes files can be unstaged despite them not having any actual conflict markers (especially in the case of a conflicting file rename). The user just needs to run `git add .`. I figured I'd add a check for conflict markers so the user could easily tell if there were any manual file changes they needed to make.